File: backend/src/routers/reports.py
"""Router de reportes — /api/v1/reports/*."""

from fastapi import APIRouter

router = APIRouter(prefix="/api/v1/reports", tags=["Reports"])


@router.get("/")
async def list_reports():
    """Listar reportes del tenant."""
    # TODO: Implementar listado
    pass


@router.post("/generate")
async def generate_report():
    """Generar un nuevo reporte (asincrónico)."""
    # TODO: Implementar generación
    pass


@router.get("/{report_id}/download")
async def download_report(report_id: str):
    """Descargar un reporte generado."""
    # TODO: Implementar descarga
    pass

File: frontend/src/api/reports.js
import client from './client';

export const reportsAPI = {
  list: () => client.get('/api/v1/reports'),
  generate: (data) => client.post('/api/v1/reports/generate', data),
  download: (id) => client.get(`/api/v1/reports/${id}/download`, { responseType: 'blob' }),
};
